plugins/modules/ec2_ami_info.py

```diff
@@ -235,52 +241,82 @@ def list_ec2_images(ec2_client, module):
             filters['owner-id'].append(owner)
         elif owner == 'self':
             # self not a valid owner-alias filter (https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeImages.html)
-            owner_param.append(owner)
+            request_args['Owners'] = [str(owner)]
         else:
             if 'owner-alias' not in filters:
                 filters['owner-alias'] = list()
             filters['owner-alias'].append(owner)
 
-    filters = ansible_dict_to_boto3_filter_list(filters)
+    request_args['Filters'] = ansible_dict_to_boto3_filter_list(filters)
+
+    request_args = {k: v for k, v in request_args.items() if v}
+
+    return request_args
+
+
+def get_images(ec2_client, request_args):
+    try:
+        images = ec2_client.describe_images(aws_retry=True, **request_args)
+    except (ClientError, BotoCoreError) as err:
+        raise AmiInfoFailure(err, "error describing images")
+    return images
+
 
+def get_image_attribute(ec2_client, image):
     try:
-        images = ec2_client.describe_images(aws_retry=True, ImageIds=image_ids, Filters=filters, Owners=owner_param,
-                                            ExecutableUsers=executable_users)
-        images = [camel_dict_to_snake_dict(image) for image in images["Images"]]
+        launch_permissions = ec2_client.describe_image_attribute(
+            aws_retry=True, Attribute='launchPermission', ImageId=image['image_id'])
     except (ClientError, BotoCoreError) as err:
-        module.fail_json_aws(err, msg="error describing images")
+        raise AmiInfoFailure(err, "error describing image attribute")
+    return launch_permissions
+
+
+def list_ec2_images(ec2_client, module, request_args):
+
+    images = get_images(ec2_client, module, request_args)["Images"]
+    images = [camel_dict_to_snake_dict(image) for image in images]
+
     for image in images:
         try:
             image['tags'] = boto3_tag_list_to_ansible_dict(image.get('tags', []))
             if module.params.get("describe_image_attributes"):
-                launch_permissions = ec2_client.describe_image_attribute(aws_retry=True, Attribute='launchPermission',
-                                                                         ImageId=image['image_id'])['LaunchPermissions']
+                launch_permissions = get_image_attribute(ec2_client, image)['LaunchPermissions']
```

Comment:
```suggestion
                launch_permissions = get_image_attribute(ec2_client, image).get('LaunchPermissions')
```

Reply:
In this case you need to default on `[]` to avoid a `TypeError` later:

```
launch_permissions = get_image_attribute(ec2_client, image).get('LaunchPermissions', [])
```

This being said, the use of .get() may also be a bit risky and hide a real problem, like e.g if the API call return something totally unexpected.